### DISCUSSION TITLE
Dunder variables in inspector.py

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2370] [Q&A] [answered]
I'm taking a look at the source code for Pytorch Geometric, and particularly I'm currently looking at `inspector.py`. I noticed that some variables of the `Inspector` class have double underscores. What is that supposed to signify? This [post](https://dbader.org/blog/meaning-of-underscores-in-python) by Dan Bader suggests to avoid naming one's variables using dunder, so I was just wondering why they're being used here.

**Comment by rusty1s** [comment, marked as answer]
Thanks for the pointer. We do not strictly follow this guidance in PyG, and we use `_*` and `__*` interchangeably to denote private methods/variables.

**Reply by mdanb** [reply]
Actually sorry for the confusion I meant the file for Message Passing
This one has variables that are called `__*__` like `__explain__` while other variables don’t. What do both leading and following dunder signify for variables?

**Reply by rusty1s** [reply]
It just signifies that those shouldn't be touched by the user, although I agree that dunder usage should be made more consistent.
